File: CP Model/model/systemicCirculation/systemicVascular.py
def systemicVascular(Psvb, Psa, Psv, Pev, Pamv, Prmv, Pbv, Phv, Rsa, Rsvb, Revb, Ramvb, Rrmvb, Rbvb, Rhvb,
                     Csvb, Cevb, Camvb, Crmvb, Cbvb, Chvb, Vusvb, Vuevb, Vuamvb, Vurmvb, Vubvb, Vuhvb):
    """
    Systemic vascular (peripheral distribution) - 6 COMPARTMENTS like Ursino 2002
    Compartments: splanchnic, extrasplanchnic, active muscle, resting muscle, brain, heart

    Returns:
    - dPsvb: pressure derivative
    - Individual flows: Qsv, Qev, Qamv, Qrmv, Qbv, Qhv
    - Individual volumes: Vsvb, Vevb, Vamvb, Vrmvb, Vbvb, Vhvb
    - Fin: inflow from systemic arteries
    """
    # Inflow from systemic arteries
    Fin = (Psa - Psvb) / Rsa

    # Individual outflows to 6 peripheral beds (A6)
    Qsv = (Psvb - Psv) / Rsvb  # Splanchnic flow
    Qev = (Psvb - Pev) / Revb  # Extrasplanchnic flow
    Qamv = (Psvb - Pamv) / Ramvb  # Active muscle flow (leg muscles)
    Qrmv = (Psvb - Prmv) / Rrmvb  # Resting muscle flow (other muscles)
    Qbv = (Psvb - Pbv) / Rbvb  # Brain flow
    Qhv = (Psvb - Phv) / Rhvb  # Heart flow

    # Individual volumes (A7)
    Vsvb = Csvb * Psvb + Vusvb  # Splanchnic volume
    Vevb = Cevb * Psvb + Vuevb  # Extrasplanchnic volume
    Vamvb = Camvb * Psvb + Vuamvb  # Active muscle volume
    Vrmvb = Crmvb * Psvb + Vurmvb  # Resting muscle volume
    Vbvb = Cbvb * Psvb + Vubvb  # Brain volume
    Vhvb = Chvb * Psvb + Vuhvb  # Heart volume

    # Total outflow
    Fout = Qsv + Qev + Qamv + Qrmv + Qbv + Qhv

    # Total compliance (A4)
    Ctot = Csvb + Cevb + Camvb + Crmvb + Cbvb + Chvb

    # Pressure derivative (A4)
    dPsvb = (Fin - Fout) / Ctot

    return dPsvb, Qsv, Qev, Qamv, Qrmv, Qbv, Qhv, Vsvb, Vevb, Vamvb, Vrmvb, Vbvb, Vhvb, Fin

File: Systemic Circulation/systemicVascular.m
function dPsvb = systemicVascular(Psvb, Psa, Psv, Pev, Pmv, Pbv, Phv, Rsa, Rsvb, Revb, Rmvb, Rbvb, Rhvb, Csvb, Cevb, Cmvb, Cbvb, Chvb)
%P = [Psvb(P1), Psa(P2), Psv(P3), Pev(P4), Pmv(P5), Pbv(P6), Phv(P7)]
%R = [Rsa(R1), Rsp(R2), Rep(R3), Rmp(R4), Rbp(R5), Rhp(R6)]
%C = [Csp(C1), Cep(C2), Cmp(C3), Cbp(C4), Chp(C5)]
    f1 = (Psvb - Psv)/Rsvb;
    f2 = (Psvb - Pev)/Revb;
    f3 = (Psvb - Pmv)/Rmvb;
    f4 = (Psvb - Pbv)/Rbvb;
    f5 = (Psvb - Phv)/Rhvb;
    Fout = f1 + f2 + f3 + f4 + f5;
    Fin = (Psa - Psvb)/Rsa;
    Ctot = Csvb + Cevb + Cmvb + Cbvb + Chvb;
    dPsvb = (Fin - Fout)/Ctot;
end